apps/fio/tests/capacity/test_capacity.yml

```diff
@@ -0,0 +1,164 @@
+
+- name: Replace the default fio profile with user-defined profile
+  replace:
+    path: "{{ pod_yaml_alias }}"
+    regexp: "testtemplate"
+    replace: "{{ lookup('env','FIO_TEST_PROFILE') }}"
+
+- name: Replace the default I/O sample size with user-defined profile
+  replace:
+    path: "{{ pod_yaml_alias }}"
+    regexp: "testsize"
+    replace: "{{ lookup('env','FIO_SAMPLE_SIZE') }}"
+
+- name: Replace the pvc placeholder with test param in app spec
+  replace:
+    path: "{{ pod_yaml_alias }}"
+    regexp: "testclaim"
+    replace: "{{ test_name }}"
+
+- name: Replace the default I/O test duration with user-defined period
+  replace:
+    path: "{{ pod_yaml_alias }}"
+    regexp: "testduration"
+    replace: "{{ lookup('env','FIO_TESTRUN_PERIOD') }}"
+
+- name: Get controller svc
+  shell: >
+    kubectl get svc -l openebs.io/controller-service=jiva-controller-svc
+    -n litmus --no-headers -o custom-columns=:spec.clusterIP
+  args:
+    executable: /bin/bash
+  register: controller_svc
+  failed_when: controller_svc.stdout == ""
+
+- name: Get controller response at /v1/volumes
+  vars:
+    volumes_endpoint: "9501/v1/volumes"
+  uri:
+    url: "http://{{ controller_svc.stdout }}:{{ volumes_endpoint }}"
+    method: GET
+    return_content: yes
+    status_code: 200
+    headers:
+      Content-Type: "application/json"
+    body_format: json
+  register: json_response
+  until: json_response.json.data[0].readOnly == "false"
+  retries: 100
+  delay: 60
+
+- name: Deploy fio test job
+  shell: >
+    kubectl apply -f {{ pod_yaml_alias }} -n litmus
+  args:
+    executable: /bin/bash
+
+- name: Get name of fio pod
+  shell: >
+     kubectl get pods -l name=fio -n litmus
+     --no-headers
+  args:
+    executable: /bin/bash
+  register: result
+  until: "'fio' and 'Running' in result.stdout"
+  delay: 10
+  retries: 30
+
+- name: Obtain name of fio pod
+  set_fact:
+    fio_pod_name: "{{ result.stdout.split()[0] }}"
+
+- name: Check fio pod status
+  shell: >
+    kubectl get pod {{ fio_pod_name }} -n litmus
+    --no-headers -o custom-columns=:status.phase
+  args:
+    executable: /bin/bash
+  register: result
+  failed_when: "result.stdout not in ['Running', 'Succeeded']"
+
+- name: Confirm fio job completion status (poll and wait every 120 sec)
+  shell: >
+    kubectl get pod {{ fio_pod_name }} -n litmus
+    --no-headers -o custom-columns=:status.phase
+  args:
+    executable: /bin/bash
+  register: result
+  until: "'Succeeded' in result.stdout"
+  delay: 120
+  retries: 30
+
+- name: Verify the fio logs to check if run is complete w/o errors
+  shell: >
+    kubectl logs {{ fio_pod_name }} -n litmus
+    | grep -i error | cut -d ":" -f 2
+    | sort | uniq
+  args:
+    executable: /bin/bash
+  register: result
+  failed_when: result.stdout != " 0,"
+
+- name: Get the name of replica pods
+  shell: >
+    kubectl get pods -l openebs.io/replica=jiva-replica
+    -n litmus --no-headers -o name | sed "s/^.\{4\}//"
```

Comment:
will it help to use jsonpath to derive names

Reply:
No @ksatchit since we are listing the n no of pods which will be used later to delete replicas randomly. We can use templates for this case also like this 
```kubectl get pods -l openebs.io/replica=jiva-replica                                                                                            
~ 115     -n litmus --template '{{range .items}}{{.metadata.name}}{{"\n"}}{{end}}'```

Reply:
:+1: